Asterix/Main_EFC_THD.py

```diff
@@ -475,12 +497,42 @@ def correctionLoop(parameter_file,
     Model_local_dir = os.path.join(Data_dir, "Model_local") + os.path.sep
 
     # Initialize thd:
-    thd2 = instr.THD2_testbed(modelconfig,
-                              DMconfig,
-                              Coronaconfig,
-                              load_fits=True,
-                              model_dir=model_dir,
-                              Model_local_dir=Model_local_dir)
+    pup_round = OptSy.pupil(modelconfig)
+    DM1 = OptSy.deformable_mirror(modelconfig,
+                                DMconfig,
+                                Name_DM='DM1',
+                                model_dir=model_dir,
+                                Model_local_dir=Model_local_dir)
+
+    DM3 = OptSy.deformable_mirror(modelconfig,
+                                DMconfig,
+                                Name_DM='DM3',
+                                model_dir=model_dir,
+                                Model_local_dir=Model_local_dir)
+
+    # we also need to "clear" the apod plane because the THD2 is like that
+    Coronaconfig.update({'filename_instr_apod': "ClearPlane"})
+    corono = OptSy.coronagraph(modelconfig, Coronaconfig, model_dir=model_dir)
+    # and then just concatenate
+    thd2 = OptSy.Testbed([pup_round, DM1, DM3, corono],
+                                ["entrancepupil", "DM1", "DM3", "corono"])
+
+    # TODO Because the code is currently setup heavily on the
+    # 'default testbed' beeing thd2 having those elements, VS code thinks
+    # there is an error if it thinks they are not defined (although in practice
+    # they are).
+    thd2.entrancepupil = pup_round
+    thd2.DM1 = DM1
+    thd2.DM3 = DM3
+    thd2.corono = corono
+    # In practice this is done inside the Testbed initialization already !
+    # and these lines are useless and I only put them to calm
+    # To be removed when the correction and estimation class are done
+    # in which case these class will be able when these things exists or not
```

Comment:
Now the estimation and correction class exist. Can you remove these lines?

Reply:
almost !